### ISSUE TITLE
Overview of EF Core and Transactions Does Not Discuss Async

### ISSUE THREAD

**Issue by richardcox13** [dotnet/EntityFramework.Docs#3250]
In EF Framework, when using transaction scopes with awaitable operations it was necessary to pass the `TransactionScopeAsyncFlowOption .Enabled` flag.

This removed the thread affinity of the scope, so should  Task continations execute on a different thread everything would work. Because thread switches cannot be controlled _and_ tended not to happen in a development environment, this flag needed to be used even when under the debugger the exceptions would not be thrown.

In EF Core is this needed? Does `DbContext.Database.BeginTransaction()` handle the async case?

As async is the generally preferred approach it would really help if the documentation made it clear if this is not needed because many developers using EF Core will have learnt EF before Core was started let alone the current platform.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1073c238-9389-d826-a696-f1108a4f2ce0
* Version Independent ID: cf56babd-def0-d76e-8eee-e33341b6a009
* Content: [Transactions - EF Core](https://docs.microsoft.com/en-us/ef/core/saving/transactions)
* Content Source: [entity-framework/core/saving/transactions.md](https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/saving/transactions.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @roji
* Microsoft Alias: **avickers**

**Comment by roji** [comment]
@richardcox13 TransactionScopeAsyncFlowOption controls whether the ambient transaction started by TransactionScope (System.Transactions) is flowed across threads when performing async operations (with await). When you use DbContext.Database.BeginTransaction (or its async counterpart, BeginTransactionAsync), you aren't using TransactionScope at all - the two transaction management mechanisms are completely unrelated, and should not be mixed.

So to summarize... If you're using TransactionScope (System.Transactions) to manage your transactions, and are using async, then yes, you probably need to specify TransactionScopeAsyncFlowOption.Enabled. If, instead, you're using the BeginTransaction APIs, then there is no TransactionScope and TransactionScopeAsyncFlowOption.Enabled isn't needed.

**Comment by richardcox13** [comment]
@roji  this information should be in the documentation.

**Comment by roji** [comment]
@richardcox13 the (non-)relationship between System.Transactions/TransactionScope and manual transaction management via BeginTransaction is a general .NET database thing rather than an EF Core thing. But I agree making this clear on our docs could be useful.

**Comment by AndriySvyryd** [comment]
We should link to System.Transactions/TransactionScope docs
